framework/tlvf/src/src/CmduMessageTx.cpp

```diff
@@ -81,14 +81,19 @@ bool CmduMessageTx::finalize(bool swap_needed)
         return true;
 
     if (!m_cmdu_header)
-        return false; 
+        return false;
 
-    // Update vendor specific length which is seperated to 3 classes
-    for (auto class_it = m_class_vector.begin(); class_it != m_class_vector.end(); class_it++) {
-        if(!(*class_it)->isPostInitSucceeded()){
+    // validate that last tlv of the CMDU is fully initialized
+    if(m_class_vector.size() != 0) {
```

Comment:
Same remarks:

- no need to check size, just check if last_tlv is valid;
- formatting is not OK.